server/src/routes/project.ts

```diff
@@ -0,0 +1,54 @@
+import { Request, Response } from "express";
+import { Project } from ".prisma/client";
+
+import { asyncHandler } from "../utils/asyncHandler";
+import { prisma } from "../common";
+import express from "express";
+
+export const projectRoutes = express.Router();
+
+projectRoutes.route("/").get(
+  asyncHandler(async (req: Request | any, res: Response) => {
+    // note: query params default to strings (e.g. 1 → '1'),
+    // thus numeric fields must be converted to numbers in
+    // order for prisma's queries to function as expected
+    Object.keys(req.query).forEach(key => {
+      if (["expo", "round", "table"].includes(key))
+        req.query[key] = parseInt(req.query[key]);
+    });
+    const matches: Project[] = await prisma.project.findMany({
+      where: req.query
+    });
+    res.status(200).send(matches);
+  })
+);
+
+projectRoutes.route("/").post(
+  asyncHandler(async (req: Request, res: Response) => {
+    const created: Project = await prisma.project.create({
+      data: req.body
+    });
+    res.status(201).json(created);
+  })
+);
+
+projectRoutes.route("/action/move").post(
+  asyncHandler(async (req: Request, res: Response) => {
+    const { ids, ...updates } = req.body;
+    const batchPayload: object = await prisma.project.updateMany({
+      where: { id: { in: ids }},
+      data: updates
```

Comment:
Oh wait, you took out the checking to make sure only those 4 fields were updated? Like tablegroup, expo, etc. Can you add those back in as a precaution?

Reply:
Discussed over Slack. Concluded that these checks aren't necessary because our API isn't publicly accessible, and thus we can assume all data passed through the request body meets our specifications. Also changed endpoint route from `/projects/action/move` to `/projects/batch/update`.